lib/knapsack_pro/config/temp_files.rb

```diff
@@ -0,0 +1,46 @@
+module KnapsackPro
+  module Config
+    class TempFiles
+      def self.temp_directory_path
+        temp_files = new
+        temp_files.ensure_temp_directory_exists
+        temp_files.temp_directory_path
+      end
+
+      def ensure_temp_directory_exists
```

Comment:
When I move code to `private` section:

```ruby
private

      def ensure_temp_directory_exists
        unless File.exists?(gitignore_file_path)
          create_temp_directory
          create_gitignore_file
        end
      end

      # relative to the directory where you run knapsack_pro gem (user's project)
      def temp_directory_path
        '.knapsack_pro'
      end
```

then I get error:

```
$ rspec ./spec/knapsack_pro/config/temp_files_spec.rb

Randomized with seed 3043

KnapsackPro::Config::TempFiles
  .temp_directory_path
    .gitignore file has content (FAILED - 1)
    creates .gitignore file (FAILED - 2)
    returns temporary directory path (FAILED - 3)

Failures:

  1) KnapsackPro::Config::TempFiles.temp_directory_path .gitignore file has content
     Failure/Error: temp_files.ensure_temp_directory_exists

     NoMethodError:
       private method `ensure_temp_directory_exists' called for #<KnapsackPro::Config::TempFiles:0x00007f911c9ed6d0>
     # ./lib/knapsack_pro/config/temp_files.rb:8:in `temp_directory_path'
     # ./spec/knapsack_pro/config/temp_files_spec.rb:5:in `block (3 levels) in <top (required)>'
     # ./spec/knapsack_pro/config/temp_files_spec.rb:18:in `block (3 levels) in <top (required)>'
     # /Users/artur/.rvm/gems/ruby-3.0.2/gems/webmock-3.13.0/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
```

---

When I move code to `protected` I get error as well:

```
  1) KnapsackPro::Config::TempFiles.temp_directory_path creates .gitignore file
     Failure/Error: temp_files.ensure_temp_directory_exists

     NoMethodError:
       protected method `ensure_temp_directory_exists' called for #<KnapsackPro::Config::TempFiles:0x00007fbb43a47c88>
     # ./lib/knapsack_pro/config/temp_files.rb:8:in `temp_directory_path'
     # ./spec/knapsack_pro/config/temp_files_spec.rb:5:in `block (3 levels) in <top (required)>'
     # ./spec/knapsack_pro/config/temp_files_spec.rb:13:in `block (3 levels) in <top (required)>'
     # /Users/artur/.rvm/gems/ruby-3.0.2/gems/webmock-3.13.0/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
```

We call the following methods in class method `temp_directory_path` so they probably must be public.

```ruby
 temp_files.ensure_temp_directory_exists
 temp_files.temp_directory_path
```

Reply:
Yes, this is because we are trying to access the private method from outside the object.

If we are adamant about the public method remaining a class method, we could change all the other methods to be [private class methods](https://medium.com/rubycademy/3-ways-to-make-class-methods-private-in-ruby-64b970e54613).